### PR TITLE
Added 4 new Java 11 recipes

### PR DESCRIPTION
**Pull request by cjobinabo**
## What's changed?

Screenshot 2023-09-19 at 5 06 49 PM
Screenshot 2023-09-19 at 5 06 27 PM
Screenshot 2023-09-19 at 5 06 10 PM
Screenshot 2023-09-19 at 5 05 53 PM

## Anything in particular you'd like reviewers to focus on?

There seems to still be issues with the `testJREDoNotUseSunNetSslInternalSslProvider` and `testJREDoNotUseSunNetSslInternalWwwProtocolHttpsHandler` tests. It seems like the dependencies provided in `sun.internal.jar` aren't resolving properly despite the code compiling properly in a live environment using that jar. Feel free to look into that issue with me.

## Anyone you would like to review specifically?

@timtebeek @joanvr

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've added the license header to any new files through `./gradlew licenseFormat`
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/resources/META-INF/rewrite/java-version-11.yml [hunk 1]
```diff
@@ -196,4 +200,62 @@ recipeList:
...
+name: org.openrewrite.java.migrate.JREDoNotUseSunNetSslInternalWwwProtocolHttpsHandler
+displayName: Use `com.ibm.net.ssl.www2.protocol.https.Handler` instead of `com.sun.net.ssl.internal.www.protocol.https.Handler`
+description: Do not use the com.sun.net.ssl.internal.www.protocol.https.Handler class.
+tags:
+  - java11
+recipeList:
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.sun.net.ssl.internal.www.protocol.https.Handler
+      newFullyQualifiedTypeName: com.ibm.net.ssl.www2.protocol.https.Handler
+      ignoreDefinition: true
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/java-version-11.yml:226]: I'm doubting if we should migrate folks over to the ibm classes; would those not be constrained to a particular JVM vendor or dependency?

Curious to hear your thoughts as I might lack some background.

Comment (cjobinabo) [reply src/main/resources/META-INF/rewrite/java-version-11.yml:226]: This is one of the recipes I mentioned would be IBM-specific. It's possible other solutions exist, but this is one we provided an alternative for. This recipe would really be aimed at our customers. If you want I can remove the recipe from the general Java 11 list.

Comment (timtebeek) [reply src/main/resources/META-INF/rewrite/java-version-11.yml:226]: I think removing it from the java-version-11.yml would be best indeed; would it fit in better with rewrite-liberty perhaps? Or would you think a separate IBM, OpenJ9 or Semuru yml file would fit in best here?

Comment (cjobinabo) [reply src/main/resources/META-INF/rewrite/java-version-11.yml:226]: A separate ibm-java yml would be a better fit.

Comment (cjobinabo) [reply src/main/resources/META-INF/rewrite/java-version-11.yml:226]: I'll push that update soon. I hadn't gotten a chance to revisit this PR until today. I just noticed that the build is passing now. Thanks for helping me work out those issues.

**Comment by cjobinabo** [comment]
For whatever reason, the 2 tests I mentioned in the description are still failing locally. It seems like there is something about my environment I'll need to fix. That is something I'll need to investigate further later.

File: src/main/resources/META-INF/rewrite/java-version-11.yml [hunk 2]
```diff
@@ -196,4 +197,27 @@ recipeList:
       groupPattern: com.sun.xml.ws
       artifactPattern: "*"
       onlyIfVersionsMatch: true
-
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.InternalBindContextFactory
+displayName: Use `com.sun.xml.bind.v2.ContextFactory` instead of `com.sun.xml.internal.bind.v2.ContextFactory`
+description: Do not use the com.sun.xml.internal.bind.v2.ContextFactory class.
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/java-version-11.yml:204]:
```suggestion
description: Do not use the `com.sun.xml.internal.bind.v2.ContextFactory` class.
```

File: src/main/resources/META-INF/rewrite/java-version-11.yml [hunk 3]
```diff
@@ -196,4 +197,27 @@ recipeList:
...
+name: org.openrewrite.java.migrate.JREDoNotUseSunNetSslInternalWwwProtocolHttpsHandler
+displayName: Use `com.ibm.net.ssl.www2.protocol.https.Handler` instead of `com.sun.net.ssl.internal.www.protocol.https.Handler`
+description: Do not use the com.sun.net.ssl.internal.www.protocol.https.Handler class.
+tags:
+  - java11
+recipeList:
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.sun.net.ssl.internal.www.protocol.https.Handler
+      newFullyQualifiedTypeName: com.ibm.net.ssl.www2.protocol.https.Handler
+      ignoreDefinition: true
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/java-version-11.yml:223]: I think this one has fully moved to ibm-java.yml right?
```suggestion
```

File: src/main/resources/META-INF/rewrite/ibm-java.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+recipeList:
+  - org.openrewrite.java.migrate.JREDoNotUseSunNetSslInternalWwwProtocolHttpsHandler
+  - org.openrewrite.java.migrate.JREDoNotUseSunNetSslInternalWwwProtocol
+  - org.openrewrite.java.migrate.JREDoNotUseSunNetSslInternalSslProvider
+
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.JREDoNotUseSunNetSslInternalWwwProtocolHttpsHandler
+displayName: Use `com.ibm.net.ssl.www2.protocol.https.Handler` instead of `com.sun.net.ssl.internal.www.protocol.https.Handler`
+description: Do not use the com.sun.net.ssl.internal.www.protocol.https.Handler class.
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/ibm-java.yml:34]:
```suggestion
description: Do not use the `com.sun.net.ssl.internal.www.protocol.https.Handler` class.
```

File: src/main/resources/META-INF/rewrite/ibm-java.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+recipeList:
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.sun.net.ssl.internal.www.protocol.https.Handler
+      newFullyQualifiedTypeName: com.ibm.net.ssl.www2.protocol.https.Handler
+      ignoreDefinition: true
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.JREDoNotUseSunNetSslInternalWwwProtocol
+displayName: Use `com.ibm.net.ssl.www2.protocol` instead of `com.sun.net.ssl.internal.www.protocol`
+description: Do not use the com.sun.net.ssl.internal.www.protocol package.
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/ibm-java.yml:46]:
```suggestion
description: Do not use the `com.sun.net.ssl.internal.www.protocol` package.
```

File: src/main/resources/META-INF/rewrite/ibm-java.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  - java11
+recipeList:
+  - org.openrewrite.java.migrate.ReplaceStringLiteralValue:
+      oldLiteralValue: com.sun.net.ssl.internal.www.protocol
+      newLiteralValue: com.ibm.net.ssl.www2.protocol
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.JREDoNotUseSunNetSslInternalSslProvider
+displayName: Use `com.ibm.jsse2` instead of `com.sun.net.ssl.internal.ssl`
+description: Do not use the com.sun.net.ssl.internal.ssl.Provider class.
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/ibm-java.yml:57]:
```suggestion
description: Do not use the `com.sun.net.ssl.internal.ssl.Provider` class.
```

**Review by timtebeek — APPROVED** [review]
Nearly there I suppose.

File: src/main/resources/META-INF/rewrite/java-version-11.yml (hunk 2 above)
Comment (cjobinabo) [reply src/main/resources/META-INF/rewrite/java-version-11.yml:204]: I didn't realize descriptions could use that code syntax. I'll go ahead and forward this to other applicable descriptions.

**Comment by timtebeek** [comment]
Thanks again! :)

**Review by kunli2 — APPROVED** [review]
LGTM, thanks for contribution!

**Comment by cjobinabo** [comment]
Thanks Kun! How can I kick off a new build when your fix goes live?

**Comment by kunli2** [comment]
The fix will need a while to go live, I think we can merge this directly, and I will re-run the build ~1 hour later to make sure it's green.

**Pull request merged**